lookout/style/typos/preparation.py

```diff
@@ -98,7 +98,10 @@ def prepare_data(params: Optional[Mapping[str, Any]] = None) -> pandas.DataFrame
         raw_data_path = os.path.join(params["data_dir"], params["raw_data_filename"])
         _download_url(params["dataset_url"], raw_data_path)
 
-    data = pandas.read_csv(raw_data_path, index_col=0)
+    try:
+        data = pandas.read_csv(raw_data_path, index_col=0)
+    except UnicodeDecodeError:
+        data = pandas.read_csv(raw_data_path, compression="xz", index_col=0)
```

Comment:
it should be possible to check file extension like
```python
if raw_data_path.endswith(".xz"):
    ...
else: 
    ...
```
It is a better approach then try-except.

Reply:
Now I see your comment: https://github.com/src-d/style-analyzer/pull/668#issuecomment-470580979
Maybe you save file to temp file without `.xz` suffix?

Reply:
Thanks!

Reply:
So pandas can read xz archives, nice

Reply:
> Maybe you save the file to temporary one without `.xz` suffix?

Oh, yes, it must be it, thank you again!